Make annotation processor lombok friendly

Lost some time trying to figure out why the validations are failing with `UnsupportedOperationException: null`.  
It turns out the validator is incompatible with the Lombok.

When the order of processors is:
1. lombok processor
2. avaje-validator-generator

Some extra `ValidationAdapter`s are generated for getters, and they fail with   
```
java.lang.UnsupportedOperationException: null
        at jars//io.avaje.validation.adapter.ValidationAdapter.primitive(ValidationAdapter.java:39)
 ```
        
Changing the order won't help either, compilation simply fails:
```
[ERROR] Non accessible field .... name with no matching getter?
[ERROR] Error ValidationAdapter due to nonAccessibleField for ....
```

--- a/validator-generator/src/main/java/io/avaje/validation/generator/TypeReader.java
+++ b/validator-generator/src/main/java/io/avaje/validation/generator/TypeReader.java
@@ -129,7 +129,11 @@ final class TypeReader {
 
   private boolean includeField(Element element) {
     return !element.getModifiers().contains(Modifier.TRANSIENT)
-        && (!element.getAnnotationMirrors().isEmpty() || element.asType().toString().contains("@"));
+        && (element.getAnnotationMirrors().stream()
+                //filter lombok's suppress warnings and generated annotations
+                .filter(m -> !m.toString().contains("@java"))
+                .anyMatch(m -> !m.toString().contains("lombok"))
+            || element.asType().toString().contains("@"));
   }
 
   private void readMethod(Element element, TypeElement type, List<FieldReader> localFields) {

--- a/validator/src/main/java/io/avaje/validation/core/CoreAdapterBuilder.java
+++ b/validator/src/main/java/io/avaje/validation/core/CoreAdapterBuilder.java
@@ -25,8 +25,6 @@ import io.avaje.validation.spi.AnnotationFactory;
 
 /** Builds and caches the ValidationAdapter adapters for DValidator. */
 final class CoreAdapterBuilder {
-  @SuppressWarnings("rawtypes")
-  public static final ValidationAdapter NOOP = (type, req, propertyName) -> true;
 
   private static final Set<Class<?>> DEFAULT_GROUP = Set.of(Default.class);
   private final DValidator context;
@@ -106,7 +104,7 @@ final class CoreAdapterBuilder {
       }
     }
     // unknown annotations have noop
-    return NOOP;
+    return NoOpValidator.INSTANCE;
   }
 
   record Request(

--- a/validator/src/main/java/io/avaje/validation/core/DValidator.java
+++ b/validator/src/main/java/io/avaje/validation/core/DValidator.java
@@ -164,7 +164,7 @@ final class DValidator implements Validator, ValidationContext {
   @Override
   @SuppressWarnings("unchecked")
   public <T> ValidationAdapter<T> noop() {
-    return CoreAdapterBuilder.NOOP;
+    return NoOpValidator.INSTANCE;
   }
 
   @Override

--- a/validator/src/main/java/io/avaje/validation/core/NoOpValidator.java
+++ b/validator/src/main/java/io/avaje/validation/core/NoOpValidator.java
@@ -3,7 +3,7 @@ package io.avaje.validation.core;
 import io.avaje.validation.adapter.ValidationAdapter;
 import io.avaje.validation.adapter.ValidationRequest;
 
-class NoOpValidator implements ValidationAdapter, ValidationAdapter.Primitive {
+public class NoOpValidator implements ValidationAdapter, ValidationAdapter.Primitive {
 
   public static final NoOpValidator INSTANCE = new NoOpValidator();
 

--- a/validator/src/main/java/io/avaje/validation/core/NoOpValidator.java
+++ b/validator/src/main/java/io/avaje/validation/core/NoOpValidator.java
@@ -1,0 +1,68 @@
+package io.avaje.validation.core;
+
+import io.avaje.validation.adapter.ValidationAdapter;
+import io.avaje.validation.adapter.ValidationRequest;
+
+class NoOpValidator implements ValidationAdapter, ValidationAdapter.Primitive {
+
+  public static final NoOpValidator INSTANCE = new NoOpValidator();
+
+  @Override
+  public boolean validate(Object value, ValidationRequest req, String propertyName) {
+
+    return true;
+  }
+
+  @Override
+  public Primitive primitive() {
+    return this;
+  }
+
+  @Override
+  public boolean validate(boolean value, ValidationRequest req, String propertyName) {
+
+    return true;
+  }
+
+  @Override
+  public boolean validate(byte value, ValidationRequest req, String propertyName) {
+
+    return true;
+  }
+
+  @Override
+  public boolean validate(char value, ValidationRequest req, String propertyName) {
+
+    return true;
+  }
+
+  @Override
+  public boolean validate(double value, ValidationRequest req, String propertyName) {
+
+    return true;
+  }
+
+  @Override
+  public boolean validate(float value, ValidationRequest req, String propertyName) {
+
+    return true;
+  }
+
+  @Override
+  public boolean validate(int value, ValidationRequest req, String propertyName) {
+
+    return true;
+  }
+
+  @Override
+  public boolean validate(long value, ValidationRequest req, String propertyName) {
+
+    return true;
+  }
+
+  @Override
+  public boolean validate(short value, ValidationRequest req, String propertyName) {
+
+    return true;
+  }
+}

--- a/validator/src/main/java/io/avaje/validation/core/NoOpValidator.java
+++ b/validator/src/main/java/io/avaje/validation/core/NoOpValidator.java
@@ -3,13 +3,12 @@ package io.avaje.validation.core;
 import io.avaje.validation.adapter.ValidationAdapter;
 import io.avaje.validation.adapter.ValidationRequest;
 
-public class NoOpValidator implements ValidationAdapter, ValidationAdapter.Primitive {
+final class NoOpValidator implements ValidationAdapter, ValidationAdapter.Primitive {
 
-  public static final NoOpValidator INSTANCE = new NoOpValidator();
+  static final NoOpValidator INSTANCE = new NoOpValidator();
 
   @Override
   public boolean validate(Object value, ValidationRequest req, String propertyName) {
-
     return true;
   }
 
@@ -20,49 +19,41 @@ public class NoOpValidator implements ValidationAdapter, ValidationAdapter.Primi
 
   @Override
   public boolean validate(boolean value, ValidationRequest req, String propertyName) {
-
     return true;
   }
 
   @Override
   public boolean validate(byte value, ValidationRequest req, String propertyName) {
-
     return true;
   }
 
   @Override
   public boolean validate(char value, ValidationRequest req, String propertyName) {
-
     return true;
   }
 
   @Override
   public boolean validate(double value, ValidationRequest req, String propertyName) {
-
     return true;
   }
 
   @Override
   public boolean validate(float value, ValidationRequest req, String propertyName) {
-
     return true;
   }
 
   @Override
   public boolean validate(int value, ValidationRequest req, String propertyName) {
-
     return true;
   }
 
   @Override
   public boolean validate(long value, ValidationRequest req, String propertyName) {
-
     return true;
   }
 
   @Override
   public boolean validate(short value, ValidationRequest req, String propertyName) {
-
     return true;
   }
 }

